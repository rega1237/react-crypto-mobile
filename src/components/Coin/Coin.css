.nav-link {
  text-decoration: none;
  display: flex;
  flex-direction: column;
  align-items: center;
  text-align: center;
  padding: 0.8rem;
  justify-content: center;
  border: solid 0.5px var(--white);
}

.img-logo {
  width: 80px;
}

.coin h2 {
  color: var(--white);
  font-family: var(--lato);
  font-size: 1.1rem;
  font-weight: 800;
}