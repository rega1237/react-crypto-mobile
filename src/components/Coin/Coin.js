import PropTypes from 'prop-types';
import { NavLink } from 'react-router-dom';
import './Coin.css';

const Coin = (props) => {
  const {
    id, name, img,
  } = props;

  return (
    <NavLink className="nav-link" to={`/details/${id}`}>
      <div className="coin">
        <div className="data-container">
          <img className="img-logo" src={img} alt={`${name} Logo`} />
          <h2>{name}</h2>
        </div>
      </div>
    </NavLink>
  );
};

Coin.propTypes = {
  img: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  id: PropTypes.string.isRequired,
};

export default Coin;
