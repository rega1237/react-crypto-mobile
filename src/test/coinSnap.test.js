import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import store from '../redux/configureStore';
import Coin from '../components/Coin/Coin';

test('selected snapshot', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <MemoryRouter>
          <Coin  
            id='bitcoin'
            img='img'
            name='bitcoin'
          />
        </MemoryRouter>
      </Provider>,
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});