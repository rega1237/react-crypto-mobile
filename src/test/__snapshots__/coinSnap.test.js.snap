// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`selected snapshot 1`] = `
<a
  aria-current={null}
  className="nav-link"
  href="/details/bitcoin"
  onClick={[Function]}
>
  <div
    className="coin"
  >
    <div
      className="data-container"
    >
      <img
        alt="bitcoin Logo"
        className="img-logo"
        src="img"
      />
      <h2>
        bitcoin
      </h2>
    </div>
  </div>
</a>
`;
